build-logic/binary-compatibility/src/main/groovy/gradlebuild/EnrichedReportRenderer.groovy

```diff
@@ -0,0 +1,104 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+import me.champeau.gradle.japicmp.report.GroovyReportRenderer
+import me.champeau.gradle.japicmp.JApiCmpWorkerAction
+import me.champeau.gradle.japicmp.report.RichReportData
+
+class EnrichedReportRenderer extends GroovyReportRenderer {
+    private static acceptedChangesRegex = ~/<a href="(.+)">accepted-public-api-changes.json<\/a>/
+
+    @Override
+    void render(File htmlReportFile, RichReportData data) {
+        super.render(htmlReportFile, enrichReport(data))
+    }
+
+    /**
+     * This is super-hacky: this report is instantiated via {@code newInstance()}, within a different
+     * classloader by {@link JApiCmpWorkerAction}, so there is no way to use a
+     * normal property on the renderer instance and just set the location of the API file in it.
+     *
+     * Instead, we'll encode the path to the file in the description data field, as a link.  This is
+     * useful regardless of this renderer's needs, since now there will be a link embedded in the report
+     * to quickly open the changes file.  By then we can also regex that path out of the description
+     * in order to create a changes {@link File} for further use in this class.
+     *
+     * @param data the report data containing a description to parse
+     * @return accepted api changes reported upon, as a file
+     */
+    private static File getAcceptedApiChangesFile(RichReportData data) {
+        def matcher = data.description =~ acceptedChangesRegex
+        return new File(matcher[0][1])
+    }
+
+    private static RichReportData enrichReport(RichReportData data) {
+        String currentApiChanges = getAcceptedApiChangesFile(data).text
+        String enrichedDesc = data.description + buildFixAllButton(currentApiChanges)
+        return new RichReportData(data.reportTitle, enrichedDesc, data.violations)
+    }
+
+    private static String buildFixAllButton(String currentApiChanges) {
+        return """
+            <script type="text/javascript">
+                function getAllErrorCorrections() {
+                    var changeElements = \$(".well pre")
+                    var result = []
+                    changeElements.each(function() {
+                        result.push(JSON.parse(this.textContent));
+                    })
+                    return result;
+                }
+
+                function appendErrorCorrections() {
+                    var result = $currentApiChanges;
```

Comment:
Not sure I'm really a huge fan with executing JSON as js code. Granted, Gradle is a code execution platform by design, but this screams INJECTION ATTACK to me. Consider a method which uses `JSON.parse`: https://stackoverflow.com/questions/13515141/html-javascript-how-to-access-json-data-loaded-in-a-script-tag-with-src-set

Reply:
I thought of this, but figured this is an internal report, and we are the ones generating the JSON, so there isn't any need to be too cautious.  I suppose we could perhaps name our own methods in such a malicious way that we run a JS injection attack on ourselves when we update our own API but that seems unlikely.

But since this is already parsing the generated JSON from the report's fixes, it might as well parse the JSON from the file in the same way.